test/integration/emitter_singleton_test.ts

```diff
@@ -0,0 +1,54 @@
+import "mocha";
+
+import { emitterFor, HTTP, Mode, Message, Emitter } from "../../src";
+
+import { fixture, assertStructured } from "./emitter_factory_test";
+
+import { rejects, doesNotReject } from "assert";
+
+describe("Emitter Singleton", () => {
+  it("emit", async () => {
+    let msg: Message | unknown = undefined;
+    const fn = async (message: Message) => {
+      msg = message;
+    };
+    const emitter = emitterFor(fn, { binding: HTTP, mode: Mode.STRUCTURED });
+    Emitter.on("cloudevent", emitter);
+    fixture.emit();
+    while (msg === undefined) {
+      await new Promise((resolve) => setTimeout(resolve, 10));
+    }
+    let body: unknown = (<Message>(<unknown>msg)).body;
+    if (typeof body === "string") {
+      body = JSON.parse(body);
+    }
+    assertStructured({ ...(<any>body), ...(<Message>(<unknown>msg)).headers });
+  });
+
+  it("emitWithEnsureDelivery", async () => {
+    let msg: Message | unknown = undefined;
+    const fn = async (message: Message) => {
+      msg = message;
+    };
+    const emitter = emitterFor(fn, { binding: HTTP, mode: Mode.STRUCTURED });
+    Emitter.on("cloudevent", emitter);
+    await fixture.emit(true);
+    let body: any = (<Message>msg).body;
+    if (typeof body === "string") {
+      body = JSON.parse(body);
+    }
+    assertStructured({ ...(<any>body), ...(<Message>(<unknown>msg)).headers });
+  });
+
+  it("emitWithEnsureDeliveryError", async () => {
+    const emitter = async () => {
+      throw new Error("Not sent");
+    };
+    Emitter.on("cloudevent", emitter);
+    // Should fail with emitWithEnsureDelivery
+    await rejects(() => fixture.emit(true));
+    // Should not fail with emitWithEnsureDelivery
+    // Work locally but not on Github Actions
+    // await doesNotReject(() => fixture.emit(false));
```

Comment:
Oh I see now why `doesNotReject` is unused. But I am confused why this fails on GH Actions but not locally (confirmed locally here too).